src/Featurizers/MedianImputerFeaturizer.h

```diff
@@ -0,0 +1,190 @@
+// ----------------------------------------------------------------------
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License
+// ----------------------------------------------------------------------
+#pragma once
+
+#include "../Archive.h"
+#include "../Featurizer.h"
+#include "../Traits.h"
+
+#include "Components/ImputerTransformer.h"
+#include "Components/MedianEstimator.h"
+#include "Components/PipelineExecutionEstimatorImpl.h"
+
+namespace Microsoft {
+namespace Featurizer {
+namespace Featurizers {
+
+namespace Details {
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         MedianImputerEstimatorImpl
+///  \brief         Implementation details for the `MedianImputerEstimatorImpl`
+///
+template <typename InputT, typename TransformedT, bool InterpolateValuesV, size_t MaxNumTrainingItemsV>
+class MedianImputerEstimatorImpl : public TransformerEstimator<InputT, TransformedT> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    static_assert(Traits<InputT>::IsNullableType, "'InputT' must be a nullable type");

```

Comment:
should there also be a static_assert that nonstd::optional<TransformedT> is the same type as InputT?

Reply:
Yes, I'll add that.

Reply:
Not necessarily, as InputT might be something like `nonstd::optional<int>` or `float`, both of which are nullable types.

Reply:
From my understanding, an imputer replaces missing values to a specified value, either min, max, median or something else. In the case of min, max and median, does it make sense if somebody set InputT as `nonstd::optional<int>` and TransformedT as `float`?

Reply:
This is the scenario (input of `nonstd::optional<int>` and and output of `float`) that requires the different types. Also, this only applies to the MedianImputerFeaturizer.

If the number of inputs is odd, the middle value is selected as the Median. If the number of inputs is even, the media value is the average of the two. This is why the output type is float for int inputs.

Not all types support average, which is why the Estimator supports the `InterpolateValuesV` template parameter.